cdap-watchdog/src/main/java/io/cdap/cdap/logging/framework/CustomLogPipelineConfigProvider.java

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright © 2020 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.logging.framework;
+
+import java.io.File;
+import java.util.List;
+
+/**
+ * Interface to provide custom log pipeline configurations.
+ */
+public interface CustomLogPipelineConfigProvider {
```

Comment:
Why needed? You can unset the `PIPELINE_CONFIG_DIR` config and have the LogPipelineLoader to check if that config exist before using it.

Reply:
fixed.